modules/programs/zsh.nix

```diff
@@ -240,20 +238,46 @@ in
   };
 
   config = mkIf cfg.enable (mkMerge [
+    (mkIf (cfg.profileExtra != "") {
+      home.file."${relToDotDir ".zprofile"}".text = cfg.profileExtra;
+    })
+
+    (mkIf (cfg.loginExtra != "") {
+      home.file."${relToDotDir ".zlogin"}".text = cfg.loginExtra;
+    })
+
+    (mkIf (cfg.logoutExtra != "") {
+      home.file."${relToDotDir ".zlogout"}".text = cfg.logoutExtra;
+    })
+
+    (mkIf cfg.oh-my-zsh.enable {
+      home.file."${relToDotDir ".zshenv"}".text = ''
+        ZSH="${pkgs.oh-my-zsh}/share/oh-my-zsh";
+        ZSH_CACHE_DIR="''${XDG_CACHE_HOME:-''$HOME/.cache}/oh-my-zsh";
```

Comment:
Why are these (and `ZDOTDIR`) now written directly instead of using sessionVariables?

Reply:
Like @rycee pointed out, it seems that ZSH, ZSH_CACHE_DIR, and ZDOTDIR are used only by zsh process itself and therefore don't have to be exported (which would happen if we would use sessionVariables). I'm not 100% sure, so please correct me if I'm wrong.

Reply:
Yeah makes sense, after thinking about it for a bit I can't find anything that wouldn't work because of this.